bazarr/subtitles/refiners/anidb.py

```diff
@@ -72,36 +80,30 @@ def get_series_id(self, mappings, tvdb_series_season, tvdb_series_id, episode):
         ]
 
         if not animes:
-            return None, None
+            return None, None, None
 
         # Sort the anime by offset in ascending order
         animes.sort(key=lambda a: a.episode_offset)
 
         # Different from Tvdb, Anidb have different ids for the Parts of a season
-        anidb_id = None
-        offset = 0
-
+        anime, episode_offset = animes[0]
```

Comment:
Why do we need this? If I'm not mistaken the first iteration of the loop will always overwrite it since the first offset supposed to be 0

Reply:
This is necessary in order to set an initial value for `anidb_id` (as done in the next line).  
Because this variable now only gets overwritten within the condition in the loop, not doing this would risk `anidb_id` being undefined if the condition never fires, as some shows don't have an offset.

Admittedly, one could rewrite this as `anime, _ = animes[0]` because `episode_offset` does get rewritten immediately.

Reply:
We are adding offset 0 to all shows that doesn't have it just above.

Reply:
```
 # Enrich the collection of anime with the episode offset
        animes = [
            self.AnimeInfo(anime, int(anime.attrib.get('episodeoffset', 0)))
            for anime in mappings.findall(
                f".//anime[@tvdbid='{tvdb_series_id}'][@defaulttvdbseason='{tvdb_series_season}']"
            )
        ]```

Reply:
Ah hold on, were you referring to `episode_offset`?  
If so, then yes I agree with you; I'll replace it with `_` as mentioned above.

Reply:
I also do not see anime being used.

Reply:
> I also do not see anime variable being used?

It is being used, see the next line:
```python
anidb_id = int(anime.attrib.get('anidbid'))
               ^^^^^
```

Reply:
Yeah but i don't know why you need to assign to the first anime if the first iteration will always do this since we force offset 0 for and sort by offset ascending.

Reply:
Yeah, that's unnecessary. I'll remove the assignment.

Reply:
Removed in [745ec97](https://github.com/morpheus65535/bazarr/pull/2505/commits/745ec973ebe9847fd299483cb7db41367587111d)